client/request_test.go

```diff
@@ -0,0 +1,59 @@
+package client
+
+import (
+	"net/url"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestExpandURL(t *testing.T) {
+	baseURL := urlMustParse("https://example.com/")
+	request := Get("endpoint/{id}{?limit}").
+		Assign("id", 1).
+		Assign("limit", "treefiddy")
+
+	url, err := request.ExpandURL(baseURL)
+
+	if assert.NoError(t, err) {
+		assert.Equal(t, "https://example.com/endpoint/1?limit=treefiddy", url.String())
+	}
+}
+
+func TestExpandURLWithBadTemplate(t *testing.T) {
+	baseURL := urlMustParse("https://example.com/")
+	request := Get("endpoint/{id").
+		Assign("id", 1)
+
+	_, err := request.ExpandURL(baseURL)
+
+	assert.Error(t, err)
+}
+
+func TestExpandURLWithNoVariableAssignments(t *testing.T) {
+	baseURL := urlMustParse("https://example.com/")
+	request := Get("endpoint/{id")
+
+	_, err := request.ExpandURL(baseURL)
+
+	assert.Error(t, err)
+}
+
+func TestExpandURLWithNoBaseURL(t *testing.T) {
+	request := Get("https://example.com/endpoint/{id}").Assign("id", 1)
+
+	url, err := request.ExpandURL(nil)
+
+	if assert.NoError(t, err) {
+		assert.Equal(t, "https://example.com/endpoint/1", url.String())
+	}
+}
+
+func urlMustParse(rawurl string) *url.URL {
```

Comment:
This is not needed when you use require instead of assert

parsed, err := url.Parse(rawurl)
require.NoError(t, err)

Reply:
It is used where I dont want to test that the user can provide me a valid URL. It just felt cleaner to have a helper function